### ISSUE TITLE
ConfigMap labels are ignored when using mock KubernetesServer

### ISSUE THREAD

**Issue by wheleph** [fabric8io/kubernetes-client#1125]
Here's the simple method that fetches list of configmaps with a particular label value:

```
public List<ConfigMap> fetchConfigMaps(KubernetesClient client, String namespace, String labelKey, String labelValue) {
    ConfigMapList configMapList = client
            .configMaps()
            .inNamespace(namespace)
            .withLabel(labelKey, labelValue)
            .list();

    return configMapList.getItems();
}
```

In the unit test below one configmap has proper label and the other one has not. So I expect that the method returns 1:

```
    @Rule
    public KubernetesServer kubernetesServerSbx = new KubernetesServer(true, true);

    @Test
    public void testFetchConfigMaps() {
        KubernetesClient client = kubernetesServerSbx.getClient();

        createNamespace(client, "ns1");
        createCM(client, "ns1", "cm1", Map.of("label1", "value"));
        createCM(client, "ns1", "cm2", Map.of("label2", "value"));

        ConfigMapFetcher fetcher = new ConfigMapFetcher();

        List<ConfigMap> configMaps = fetcher.fetchConfigMaps(client, "ns1","label1", "value");

        assertEquals(1, configMaps.size());
    }

    private static void createNamespace(KubernetesClient client, String namespace) {
        client.namespaces().create(
                new NamespaceBuilder()
                        .withNewMetadata()
                        .withName(namespace)
                        .endMetadata()
                        .build());
    }

    private static void createCM(KubernetesClient client,
                                 String namespace,
                                 String cmName,
                                 Map<String, String> labels) {

        client.configMaps().inNamespace(namespace).create(
                new ConfigMapBuilder()
                        .withNewMetadata()
                        .withName(cmName)
                        .withLabels(labels)
                        .endMetadata()
                        .build());
    }
```

However the unit test fails because the method return 2 instead of 1.

I'm using `kubernetes-client` and `kubernetes-server-mock` version 3.2.0

Here's the complete source code: [my-app.zip](https://github.com/fabric8io/kubernetes-client/files/2171556/my-app.zip)

**Comment by rohanKanojia** [comment]
Closed via #1158

**Comment by wheleph** [comment]
The fix is scheduled for version 4.0.4
